### DISCUSSION TITLE
minion join event and minion disconnected status

### DISCUSSION THREAD

**Discussion by whytewolf** [saltstack/salt#67278] [Feature Requests]
### Description of the Issue
feature-request. Minions that separate from the master such as laptops, or other portable systems, will most likely rejoin the master at some point in the future. Currently, the only events that trigger this event also trigger other events and it would be good to have a definitive I'm rejoining the master event. 

Of the events that currently work with this 

* start events. Don't cover if the minion didn't actually restart to rejoin the master
* presence events only fire on the interval not when the minion rejoins. 
* Auth events fire all the time regardless of a rejoining. 

As a side effect of this event, another item that could be brought up is the ability to put minions into a disconnected state according to the master. so that they are not included in reports that target multiple minions.

**Comment by whytewolf** [comment]
zd-4393

**Comment by rico256-cn** [comment]
I have the same problem, I need a minion disconnect event to build my Application.
**When the feature will release ?**

**Comment by rico256-cn** [comment]
@waynew

**Comment by waynew** [comment]
@rico256-cn thanks for the interest! I haven't started work on this yet, if you're interested in implementing it you're more than welcome to give it a shot. PRs are welcome here! :)
